lib/common/client/middleware/response/soap_parser.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+module Common
+  module Client
+    module Middleware
+      module Response
+        class SOAPParser < Faraday::Response::Middleware
+          def on_complete(env)
+            case env.status
+            when 200
+              doc = Ox.parse(ensure_xml_prolog(env.body))
+              if doc_includes_error(doc)
+                raise Common::Client::Errors::HTTPError.new('SOAP service returned internal server error', 500)
+              end
+              env.body = doc
+            else
+              raise Common::Client::Errors::HTTPError.new('SOAP HTTP call failed', env.status)
+            end
+          end
+
+          private
+
+          def ensure_xml_prolog(xml)
+            xml = xml.dup.prepend('<?xml version="1.0" encoding="UTF-8"?>') unless xml =~ /^<\?xml/
+            xml
+          end
+
+          def doc_includes_error(doc)
```

Comment:
Based on my understanding of `doc.locate`, I think this can be made into a one-liner
```ruby
def doc_includes_error
  doc.locate('env:Envelope/env:Body/env:Fault').empty?
end
```

Also nitpicky, but I'd add a question mark to the method name to indicate it returns a `boolean`.